storage-bigtable/src/lib.rs

```diff
@@ -790,21 +790,41 @@ impl LedgerStorage {
             })
             .collect();
 
+        let mut tasks = vec![];
+
         if !tx_cells.is_empty() {
-            bytes_written += self
-                .connection
-                .put_bincode_cells_with_retry::<TransactionInfo>("tx", &tx_cells)
-                .await?;
+            let conn = self.connection.clone();
+            tasks.push(tokio::spawn(async move {
+                conn.put_bincode_cells_with_retry::<TransactionInfo>("tx", &tx_cells)
+                    .await
+            }));
         }
 
         if !tx_by_addr_cells.is_empty() {
-            bytes_written += self
-                .connection
-                .put_protobuf_cells_with_retry::<tx_by_addr::TransactionByAddr>(
+            let conn = self.connection.clone();
+            tasks.push(tokio::spawn(async move {
+                conn.put_protobuf_cells_with_retry::<tx_by_addr::TransactionByAddr>(
                     "tx-by-addr",
                     &tx_by_addr_cells,
                 )
-                .await?;
+                .await
+            }));
+        }
+
+        let results = futures::future::join_all(tasks).await;
+        let results: Vec<_> = results.into_iter().map(|r| r.unwrap()).collect();
```

Comment:
Let's map the tokio Error to something, just to be safe. Can you add an `Error` variant?
Also, can we rewrite this whole block to only iterate once?
Something like:
```
let mut bytes_written = 0;
let mut maybe_first_err: Option<Error> = None;
for result in results {
    match result {
        Err(err) => if maybe_first_err.is_none() {
            maybe_first_err = Some(Error::TokioError(err));
        }
        Ok(Err(err)) => if maybe_first_err.is_none() {
            maybe_first_err = Some(Error::BigTableError(err));
        }
        Ok(Ok(bytes)) => {
            bytes_written += bytes;
        }
    }
}
if let Some(err) = maybe_first_err {
    return Err(err);
}
```